crates/sui-keys/src/keystore.rs

```diff
@@ -145,31 +168,80 @@ impl AccountKeystore for FileBasedKeystore {
     where
         T: Serialize,
     {
-        Ok(Signature::new_secure(
-            &IntentMessage::new(intent, msg),
-            self.keys.get(address).ok_or_else(|| {
-                signature::Error::from_source(format!("Cannot find key for address: [{address}]"))
-            })?,
-        ))
+        let (_, key) = self.keys.get(address).ok_or_else(|| {
+            signature::Error::from_source(format!("Cannot find key for address: [{address}]"))
+        })?;
+        Ok(Signature::new_secure(&IntentMessage::new(intent, msg), key))
     }
 
-    fn add_key(&mut self, keypair: SuiKeyPair) -> Result<(), anyhow::Error> {
+    /// Add a key to the keystore
+    fn add_key(&mut self, alias: Option<String>, keypair: SuiKeyPair) -> Result<(), anyhow::Error> {
+        self.alias_exists(&alias)?;
+        let alias = alias.unwrap_or_else(|| random_name(self.get_aliases()));
         let address: SuiAddress = (&keypair.public()).into();
-        self.keys.insert(address, keypair);
+        self.keys.insert(address, (alias, keypair));
         self.save()?;
         Ok(())
     }
 
+    /// Return the public keys in the keystore
     fn keys(&self) -> Vec<PublicKey> {
-        self.keys.values().map(|key| key.public()).collect()
+        self.keys.values().map(|(_, key)| key.public()).collect()
     }
 
+    /// Return a list of pairs with the alias and the public key
+    fn keys_with_aliases(&self) -> Vec<(SuiAddressAlias, PublicKey)> {
+        self.keys
+            .values()
+            .map(|(alias, key)| (alias.clone(), key.public()))
+            .collect()
+    }
+
+    /// Return the corresponding SuiKeyPair of the provided sui address
     fn get_key(&self, address: &SuiAddress) -> Result<&SuiKeyPair, anyhow::Error> {
         match self.keys.get(address) {
-            Some(key) => Ok(key),
+            Some((_, key)) => Ok(key),
             None => Err(anyhow!("Cannot find key for address: [{address}]")),
         }
     }
+
+    /// Return the list of aliases in the keystore
+    fn get_aliases(&self) -> Vec<SuiAddressAlias> {
+        self.keys_with_aliases()
+            .into_iter()
+            .map(|(alias, _)| alias)
+            .collect()
+    }
+
+    /// Return an error if the provided alias exists already
+    fn alias_exists(&self, alias: &Option<String>) -> Result<(), anyhow::Error> {
+        if let Some(alias) = alias {
+            if self.get_aliases().contains(alias) {
+                anyhow::bail!("Alias {alias} already exists. Please choose another alias.")
+            }
+        }
+        Ok(())
+    }
+
+    /// Update the old alias to the new alias
+    fn update_alias(&mut self, old_alias: &str, new_alias: &str) -> Result<(), anyhow::Error> {
+        if !self.get_aliases().contains(&old_alias.to_string()) {
+            anyhow::bail!("Alias {old_alias} does not exist in keystore.")
+        } else {
+            self.keys.values_mut().for_each(|x| {
```

Comment:
This command is missing two tests:

- That the new alias does not equal the old alias (which means this is a nop)
- That the new alias does not exist.